### ISSUE TITLE
Reduce log message noise

### ISSUE THREAD

**Issue by samsrabin** [ESCOMP/CTSM#2736]
As discussed at [CSEG meeting 2024-06-18](https://docs.google.com/document/d/186U6-dt_wWZZGU9NzYQ5zNlMnpx9XX6oweuTXzQY-oo/edit#heading=h.gjot13e7ggkb), reducing unnecessary messaging is a CSEG goal for CESM3.

Things to consider for each message:
- Is it needed at all?
- Can it be changed to only show if the XML `DEBUG` setting is true? (By wrapping in `#ifndef NDEBUG`/`#endif`)
- If it's currently in cesm.log, should it be moved to lnd.log?
- If every process is printing it, can it be set so that just the primary process does?

Messages to consider:
- [x] `clm: leaving fates model` (lnd.log, **every process**)—especially strange considering there's no "entering" message.  _Get rid of this_
- [x] `CN Soil matrix solution is off` (**cesm.log**, **every process**) _get rid of this_ (also filed at #2717)
- ~~`pes participating in computation for CLM` and table (lnd.log),~~ _maybe we keep this?_ **yes**
- [x] `Init2: calling FireInit` (**cesm.log**, **every process**)

**Comment by samsrabin** [comment]
Seems like this could be accomplished like so:
https://github.com/ESCOMP/CTSM/blob/0999a32f520c995c3dfc94b9b96781d20ed0d6d5/src/biogeophys/WaterStateType.F90#L883-L885

But I'm not sure it's even necessary in the debug case.

**Comment by jedwards4b** [comment]
Noting that this still has not been addressed as of ctsm5.3.014

**Comment by wwieder** [comment]
@samsrabin is this something that can be easily addressed and put on b4b-dev?

**Comment by samsrabin** [comment]
Yes, definitely. However, I'll need to wait until b4b-dev gets the eventual fix to the Izumi nag failures introduced (probably) with the Bytnerowicz update.

**Comment by wwieder** [comment]
Makes sense. Seems like the nag bug will be addressed this week

**Comment by ekluzek** [comment]
@wwieder and @samsrabin we just merged the fix to b4b-dev. So b4b-dev is ready to have PR's added to it again.
